test/system/flows/profile/user_views_published_solutions_test.rb

```diff
@@ -23,6 +23,7 @@ class UserViewsPublishedSolutionsTest < ApplicationSystemTestCase
     end
 
     test "shows published solutions for a logged out user" do
+      User::Profile.stubs(:count_to_show_tab).returns(1)
```

Comment:
I think it's easier to do this than adding all the extra code to profile:

```suggestion
      User::Profile.any_instance.stubs(:solutions_tab?).returns(true)
```

Reply:
Okay. I'll just do it this way and add a TODO to change this. This isn't really a good thing to stub because it's not a direct value.

Reply:
I think in this case, because we're testing that the user is redirected if the `solutions_tab` logic returns false, it's ok to stub that logic out. We're not actually testing the logic of `solutions_tab` itself here, just what happens when the solutions tab shouldn't show. By doing this, if we make that logic more complex in the future (which is quite possible), this test doesn't need to be updated to respect the new rules. This is why I'm a fan of stubbing public methods like this, and then making sure all public methods have unit tests, because it means the tests need changing a lot less often.

Reply:
It's a system test. You shouldn't stub methods.

Reply:
I agree. In which case, don't stub anything and just create the relevant db records :) 

(I'm not suggesting we stub something here, just that we stub less underlying logic than what you stubbed :P)

Reply:
The reason I care is that I don't like going from one quite clean method (`if x > 3`) to having that method, plus a constant, pls a class method. And doing that twice once for solutions and once for testimonials. That would add an extra 14 LOCs in total, which just makes the code so much harder to parse when you're reading through trying to find stuff later. 

I'm ok with it more in commands, where there's less code in general (e.g where you've done it for pagination) but in models, which can get big and hairy anyway, moving from a three line method to ten lines, for no functional improvement, doesn't feel like a tradeoff I'm happy with.